src/wp-includes/block-template-utils.php

```diff
@@ -957,34 +1014,10 @@ function get_block_templates( $query = array(), $template_type = 'wp_template' )
 	}
```

Comment:
While we are at it can we change these lines. 


```
	if ( isset( $query['slug__in'] ) ) {
		$wp_query_args['post_name__in'] = $query['slug__in'];
	}
```

To 
```
	if ( isset( $query['slug__in'] ) ) {
		$wp_query_args['post_name__in'] = $query['slug__in'];
               $wp_query_args['posts_per_page'] = count( $query['slug__in'] );
	}
```

This limits the query and improves performance.

Reply:
Done at https://github.com/WordPress/wordpress-develop/pull/4097/commits/8a0f1f958ee348ef29fe8e975203db9a33a1a736